core/lxdprofile/validate.go

```diff
@@ -45,3 +45,13 @@ func ValidateCharmInfoLXDProfile(info *apicharms.CharmInfo) error {
 	}
 	return nil
 }
+
+// IsEmpty will return true if the profiler containers a profile, that is
+// empty. If the profile isn't empty, we'll return false.
+// If there is no valid profile in the profiler, it will return true
+func IsEmpty(profiler charm.LXDProfiler) bool {
```

Comment:
Suggest rename to NotEmpty and reverse the logic.  We'll always be doing !lxdprofile.IsEmpty() I think.

This will be a nice addition.  Will have to update the other bits of code. I had to go searching for the line: if profile := profiler.LXDProfile(); profile != nil && !profile.Empty() the other day to remember the combo.

Reply:
Will do